### ISSUE TITLE
tests fail in Fedora Rawhide (python 3.5)

### ISSUE THREAD

**Issue by maxamillion** [projectatomic/osbs-client#325]
```
=================================== FAILURES ===================================
 test_get_time_from_rfc3339_valid_alt_format[2015-08-24T10:41:00.1Z-1440412860.1-UTC] 
rfc3339 = '2015-08-24T10:41:00.1Z', seconds = 1440412860.1, tz = 'UTC'
    @pytest.mark.skipif(sys.version_info[0] < 3,
                        reason="requires python3")
    @pytest.mark.parametrize('tz', [
        'UTC',
        'EST',
    ])
    @pytest.mark.parametrize(('rfc3339', 'seconds'), [
        # These tests only work in Python 3
        ('2015-08-24T10:41:00.1Z', 1440412860.1),
        ('2015-09-22T11:12:00+01:00', 1442916720),
    ])
    def test_get_time_from_rfc3339_valid_alt_format(rfc3339, seconds, tz):
        os.environ['TZ'] = tz
        tzset()
>       assert get_time_from_rfc3339(rfc3339) == seconds
tests/test_utils.py:95: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
osbs/utils.py:282: in get_time_from_rfc3339
    time_tuple = strptime(rfc3339, '%Y-%m-%dT%H:%M:%SZ')
/usr/lib64/python3.5/_strptime.py:494: in _strptime_time
    tt = _strptime(data_string, format)[0]
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
data_string = '2015-08-24T10:41:00.1Z', format = '%Y-%m-%dT%H:%M:%SZ'
    def _strptime(data_string, format="%a %b %d %H:%M:%S %Y"):
        """Return a 2-tuple consisting of a time struct and an int containing
        the number of microseconds based on the input string and the
        format string."""
    
        for index, arg in enumerate([data_string, format]):
            if not isinstance(arg, str):
                msg = "strptime() argument {} must be str, not {}"
                raise TypeError(msg.format(index, type(arg)))
    
        global _TimeRE_cache, _regex_cache
        with _cache_lock:
    
            if _getlang() != _TimeRE_cache.locale_time.lang:
                _TimeRE_cache = TimeRE()
                _regex_cache.clear()
            if len(_regex_cache) > _CACHE_MAX_SIZE:
                _regex_cache.clear()
            locale_time = _TimeRE_cache.locale_time
            format_regex = _regex_cache.get(format)
            if not format_regex:
                try:
                    format_regex = _TimeRE_cache.compile(format)
                # KeyError raised when a bad format is found; can be specified as
                # \\, in which case it was a stray % but with a space after it
                except KeyError as err:
                    bad_directive = err.args[0]
                    if bad_directive == "\\":
                        bad_directive = "%"
                    del err
                    raise ValueError("'%s' is a bad directive in format '%s'" %
                                        (bad_directive, format)) from None
                # IndexError only occurs when the format string is "%"
                except IndexError:
                    raise ValueError("stray %% in format '%s'" % format) from None
                _regex_cache[format] = format_regex
        found = format_regex.match(data_string)
        if not found:
            raise ValueError("time data %r does not match format %r" %
>                            (data_string, format))
E           ValueError: time data '2015-08-24T10:41:00.1Z' does not match format '%Y-%m-%dT%H:%M:%SZ'
/usr/lib64/python3.5/_strptime.py:337: ValueError
 test_get_time_from_rfc3339_valid_alt_format[2015-08-24T10:41:00.1Z-1440412860.1-EST] 
rfc3339 = '2015-08-24T10:41:00.1Z', seconds = 1440412860.1, tz = 'EST'
    @pytest.mark.skipif(sys.version_info[0] < 3,
                        reason="requires python3")
    @pytest.mark.parametrize('tz', [
        'UTC',
        'EST',
    ])
    @pytest.mark.parametrize(('rfc3339', 'seconds'), [
        # These tests only work in Python 3
        ('2015-08-24T10:41:00.1Z', 1440412860.1),
        ('2015-09-22T11:12:00+01:00', 1442916720),
    ])
    def test_get_time_from_rfc3339_valid_alt_format(rfc3339, seconds, tz):
        os.environ['TZ'] = tz
        tzset()
>       assert get_time_from_rfc3339(rfc3339) == seconds
tests/test_utils.py:95: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
osbs/utils.py:282: in get_time_from_rfc3339
    time_tuple = strptime(rfc3339, '%Y-%m-%dT%H:%M:%SZ')
/usr/lib64/python3.5/_strptime.py:494: in _strptime_time
    tt = _strptime(data_string, format)[0]
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
data_string = '2015-08-24T10:41:00.1Z', format = '%Y-%m-%dT%H:%M:%SZ'
    def _strptime(data_string, format="%a %b %d %H:%M:%S %Y"):
        """Return a 2-tuple consisting of a time struct and an int containing
        the number of microseconds based on the input string and the
        format string."""
    
        for index, arg in enumerate([data_string, format]):
            if not isinstance(arg, str):
                msg = "strptime() argument {} must be str, not {}"
                raise TypeError(msg.format(index, type(arg)))
    
        global _TimeRE_cache, _regex_cache
        with _cache_lock:
    
            if _getlang() != _TimeRE_cache.locale_time.lang:
                _TimeRE_cache = TimeRE()
                _regex_cache.clear()
            if len(_regex_cache) > _CACHE_MAX_SIZE:
                _regex_cache.clear()
            locale_time = _TimeRE_cache.locale_time
            format_regex = _regex_cache.get(format)
            if not format_regex:
                try:
                    format_regex = _TimeRE_cache.compile(format)
                # KeyError raised when a bad format is found; can be specified as
                # \\, in which case it was a stray % but with a space after it
                except KeyError as err:
                    bad_directive = err.args[0]
                    if bad_directive == "\\":
                        bad_directive = "%"
                    del err
                    raise ValueError("'%s' is a bad directive in format '%s'" %
                                        (bad_directive, format)) from None
                # IndexError only occurs when the format string is "%"
                except IndexError:
                    raise ValueError("stray %% in format '%s'" % format) from None
                _regex_cache[format] = format_regex
        found = format_regex.match(data_string)
        if not found:
            raise ValueError("time data %r does not match format %r" %
>                            (data_string, format))
E           ValueError: time data '2015-08-24T10:41:00.1Z' does not match format '%Y-%m-%dT%H:%M:%SZ'
/usr/lib64/python3.5/_strptime.py:337: ValueError
 test_get_time_from_rfc3339_valid_alt_format[2015-09-22T11:12:00+01:00-1442916720-UTC] 
rfc3339 = '2015-09-22T11:12:00+01:00', seconds = 1442916720, tz = 'UTC'
    @pytest.mark.skipif(sys.version_info[0] < 3,
                        reason="requires python3")
    @pytest.mark.parametrize('tz', [
        'UTC',
        'EST',
    ])
    @pytest.mark.parametrize(('rfc3339', 'seconds'), [
        # These tests only work in Python 3
        ('2015-08-24T10:41:00.1Z', 1440412860.1),
        ('2015-09-22T11:12:00+01:00', 1442916720),
    ])
    def test_get_time_from_rfc3339_valid_alt_format(rfc3339, seconds, tz):
        os.environ['TZ'] = tz
        tzset()
>       assert get_time_from_rfc3339(rfc3339) == seconds
tests/test_utils.py:95: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
osbs/utils.py:282: in get_time_from_rfc3339
    time_tuple = strptime(rfc3339, '%Y-%m-%dT%H:%M:%SZ')
/usr/lib64/python3.5/_strptime.py:494: in _strptime_time
    tt = _strptime(data_string, format)[0]
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
data_string = '2015-09-22T11:12:00+01:00', format = '%Y-%m-%dT%H:%M:%SZ'
    def _strptime(data_string, format="%a %b %d %H:%M:%S %Y"):
        """Return a 2-tuple consisting of a time struct and an int containing
        the number of microseconds based on the input string and the
        format string."""
    
        for index, arg in enumerate([data_string, format]):
            if not isinstance(arg, str):
                msg = "strptime() argument {} must be str, not {}"
                raise TypeError(msg.format(index, type(arg)))
    
        global _TimeRE_cache, _regex_cache
        with _cache_lock:
    
            if _getlang() != _TimeRE_cache.locale_time.lang:
                _TimeRE_cache = TimeRE()
                _regex_cache.clear()
            if len(_regex_cache) > _CACHE_MAX_SIZE:
                _regex_cache.clear()
            locale_time = _TimeRE_cache.locale_time
            format_regex = _regex_cache.get(format)
            if not format_regex:
                try:
                    format_regex = _TimeRE_cache.compile(format)
                # KeyError raised when a bad format is found; can be specified as
                # \\, in which case it was a stray % but with a space after it
                except KeyError as err:
                    bad_directive = err.args[0]
                    if bad_directive == "\\":
                        bad_directive = "%"
                    del err
                    raise ValueError("'%s' is a bad directive in format '%s'" %
                                        (bad_directive, format)) from None
                # IndexError only occurs when the format string is "%"
                except IndexError:
                    raise ValueError("stray %% in format '%s'" % format) from None
                _regex_cache[format] = format_regex
        found = format_regex.match(data_string)
        if not found:
            raise ValueError("time data %r does not match format %r" %
>                            (data_string, format))
E           ValueError: time data '2015-09-22T11:12:00+01:00' does not match format '%Y-%m-%dT%H:%M:%SZ'
/usr/lib64/python3.5/_strptime.py:337: ValueError
 test_get_time_from_rfc3339_valid_alt_format[2015-09-22T11:12:00+01:00-1442916720-EST] 
rfc3339 = '2015-09-22T11:12:00+01:00', seconds = 1442916720, tz = 'EST'
    @pytest.mark.skipif(sys.version_info[0] < 3,
                        reason="requires python3")
    @pytest.mark.parametrize('tz', [
        'UTC',
        'EST',
    ])
    @pytest.mark.parametrize(('rfc3339', 'seconds'), [
        # These tests only work in Python 3
        ('2015-08-24T10:41:00.1Z', 1440412860.1),
        ('2015-09-22T11:12:00+01:00', 1442916720),
    ])
    def test_get_time_from_rfc3339_valid_alt_format(rfc3339, seconds, tz):
        os.environ['TZ'] = tz
        tzset()
>       assert get_time_from_rfc3339(rfc3339) == seconds
tests/test_utils.py:95: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
osbs/utils.py:282: in get_time_from_rfc3339
    time_tuple = strptime(rfc3339, '%Y-%m-%dT%H:%M:%SZ')
/usr/lib64/python3.5/_strptime.py:494: in _strptime_time
    tt = _strptime(data_string, format)[0]
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
data_string = '2015-09-22T11:12:00+01:00', format = '%Y-%m-%dT%H:%M:%SZ'
    def _strptime(data_string, format="%a %b %d %H:%M:%S %Y"):
        """Return a 2-tuple consisting of a time struct and an int containing
        the number of microseconds based on the input string and the
        format string."""
    
        for index, arg in enumerate([data_string, format]):
            if not isinstance(arg, str):
                msg = "strptime() argument {} must be str, not {}"
                raise TypeError(msg.format(index, type(arg)))
    
        global _TimeRE_cache, _regex_cache
        with _cache_lock:
    
            if _getlang() != _TimeRE_cache.locale_time.lang:
                _TimeRE_cache = TimeRE()
                _regex_cache.clear()
            if len(_regex_cache) > _CACHE_MAX_SIZE:
                _regex_cache.clear()
            locale_time = _TimeRE_cache.locale_time
            format_regex = _regex_cache.get(format)
            if not format_regex:
                try:
                    format_regex = _TimeRE_cache.compile(format)
                # KeyError raised when a bad format is found; can be specified as
                # \\, in which case it was a stray % but with a space after it
                except KeyError as err:
                    bad_directive = err.args[0]
                    if bad_directive == "\\":
                        bad_directive = "%"
                    del err
                    raise ValueError("'%s' is a bad directive in format '%s'" %
                                        (bad_directive, format)) from None
                # IndexError only occurs when the format string is "%"
                except IndexError:
                    raise ValueError("stray %% in format '%s'" % format) from None
                _regex_cache[format] = format_regex
        found = format_regex.match(data_string)
        if not found:
            raise ValueError("time data %r does not match format %r" %
>                            (data_string, format))
E           ValueError: time data '2015-09-22T11:12:00+01:00' does not match format '%Y-%m-%dT%H:%M:%SZ'
/usr/lib64/python3.5/_strptime.py:337: ValueError
===================== 4 failed, 167 passed in 9.39 seconds =====================
error: Bad exit status from /var/tmp/rpm-tmp.l9bTRC (%check)
RPM build errors:
    Bad exit status from /var/tmp/rpm-tmp.l9bTRC (%check)
Child return code was: 1
EXCEPTION: Command failed. See logs for output.
 # bash --login -c /usr/bin/rpmbuild -bb --target x86_64 --nodeps  /builddir/build/SPECS/osbs-client.spec 
Traceback (most recent call last):
  File "/usr/lib/python2.7/site-packages/mockbuild/trace_decorator.py", line 84, in trace
    result = func(*args, **kw)
  File "/usr/lib/python2.7/site-packages/mockbuild/util.py", line 493, in do
    raise exception.Error("Command failed. See logs for output.\n # %s" % (command,), child.returncode)
Error: Command failed. See logs for output.
 # bash --login -c /usr/bin/rpmbuild -bb --target x86_64 --nodeps  /builddir/build/SPECS/osbs-client.spec 
LEAVE do --> EXCEPTION RAISED
```

**Comment by twaugh** [comment]
This happens because that test only runs for Python 3, knowing that `dateutil` will understand what to do. When it isn't installed, we fall back to the `strptime` approach to parsing dates which doesn't handle fractional seconds (because OpenShift doesn't currently produce them).

The answer is to install `python3-dateutil` when running tests under Python 3. I've submitted a PR to fix the spec file.

**Comment by maxamillion** [comment]
@twaugh awesome, thanks!

**Comment by twaugh** [comment]
Thanks for reporting it!
Fixed via #326.
